OsmAnd/src/net/osmand/plus/configmap/tracks/TracksComparator.java

```diff
@@ -135,15 +135,33 @@ private int compareTrackItems(@NonNull TrackItem item1, @NonNull TrackItem item2
 			case NAME_DESCENDING:
 				return -compareTrackItemNames(item1, item2);
 			case DATE_ASCENDING:
-				if (analysis1.startTime == analysis2.startTime) {
+				long startTime1_asc = analysis1 == null ? 0 : analysis1.startTime;
+				long startTime2_asc = analysis2 == null ? 0 : analysis2.startTime;
+				long time1_asc = dataItem1 == null ? startTime1_asc : dataItem1.getFileCreationTime();
+				long time2_asc = dataItem2 == null ? startTime2_asc : dataItem2.getFileCreationTime();
+				if (time1_asc == time2_asc || time1_asc < 10 && time2_asc < 10) {
 					return compareTrackItemNames(item1, item2);
 				}
-				return -Long.compare(analysis1.startTime, analysis2.startTime);
+				if (time1_asc < 10) {
+					return 1;
+				} else if (time2_asc < 10) {
+					return -1;
+				}
+				return -Long.compare(time1_asc, time2_asc);
 			case DATE_DESCENDING:
-				if (analysis1.startTime == analysis2.startTime) {
+				long startTime1_desc = analysis1 == null ? 0 : analysis1.startTime;
+				long startTime2_desc = analysis2 == null ? 0 : analysis2.startTime;
+				long time1_desc = dataItem1 == null ? startTime1_desc : dataItem1.getFileCreationTime();
+				long time2_desc = dataItem2 == null ? startTime2_desc : dataItem2.getFileCreationTime();
+				if (time1_desc == time2_desc || time1_desc < 10 && time2_desc < 10) {
 					return compareTrackItemNames(item1, item2);
 				}
-				return Long.compare(analysis1.startTime, analysis2.startTime);
+				if (time1_desc < 10) {
+					return 1;
```

Comment:
time1_asc < 10 returns 1 and time1_desc < 10 returns 1
Is it correct?

Reply:
yes, they are in different branches and actually do the same. it means that tracks without creating time should go to bottom. in real life there should be no such tracks